### PR TITLE
Allow single and double visualization to be overridden

### PR DESCRIPTION
**Pull request by 333fred**

### CODE REVIEW

**Comment by tmat** [comment]
Looks good, but what is it useful for?

File: src/Microsoft.Metadata.Visualizer/ILVisualizer.cs [hunk 1]
```diff
@@ -417,11 +431,33 @@ public void VisualizeHeader(StringBuilder sb, int codeSize, int maxStack, Immuta
...
+                            {
+                                var value = ReadSingle(ilBytes, ref curIndex);
+                                if (value == 0 && 1 / value < 0)
+                                {
+                                    sb.Append(" -0.0");
+                                }
+                                else
+                                {
+                                    sb.Append(" ");
+                                    sb.Append(VisualizeSingle(value));
```
Comment (tmat) [inline src/Microsoft.Metadata.Visualizer/ILVisualizer.cs]: This is odd - to be able to override the behavior for some values but not others. Why not move this logic to `VisualizeSingle()`?

File: src/Microsoft.Metadata.Visualizer/ILVisualizer.cs [hunk 2]
```diff
@@ -129,6 +129,8 @@ public LocalInfo(string name, object type, bool isPinned, bool isByRef)
         public virtual string VisualizeUserString(uint token) => $"0x{token:X8}";
         public virtual string VisualizeSymbol(uint token, OperandType operandType) => $"0x{token:X8}";
         public virtual string VisualizeLocalType(object type) => $"0x{type:X8}";
+        public virtual string VisualizeSingle(float single) => $"{single:G7}";
+        public virtual string VisualizeDouble(double @double) => $"{@double:G15}";
```
Comment (tmat) [inline src/Microsoft.Metadata.Visualizer/ILVisualizer.cs]: Why not name the parameter `value`?

Comment (333fred) [reply src/Microsoft.Metadata.Visualizer/ILVisualizer.cs]: I certainly could, this is just what made sense to me.

File: src/Microsoft.Metadata.Visualizer/ILVisualizer.cs [hunk 3]
```diff
@@ -129,6 +129,15 @@ public LocalInfo(string name, object type, bool isPinned, bool isByRef)
         public virtual string VisualizeUserString(uint token) => $"0x{token:X8}";
         public virtual string VisualizeSymbol(uint token, OperandType operandType) => $"0x{token:X8}";
         public virtual string VisualizeLocalType(object type) => $"0x{type:X8}";
+        public virtual string VisualizeSingle(float single) 
+            => (single == 0 && 1 / single < 0)
+               ? "-0.0"
+               : $"{single:G15}";
```
Comment (jaredpar) [inline src/Microsoft.Metadata.Visualizer/ILVisualizer.cs]: Why G15 instead of G7?

**Pull request merged**